src/tests/functional/navigation_tests.ts

```diff
@@ -116,6 +125,33 @@ export class NavigationTests extends TurboDriveTestCase {
     this.assert.equal(await this.pathname, "/src/tests/fixtures/one.html")
     this.assert.equal(await this.visitAction, "restore")
   }
+
+  async "test skip link with hash-only path"() {
+    this.listenForVisits()
+    await this.clickSelector('a[href="#main"]')
+
+    this.assert.notOk(this.turboVisited)
```

Comment:
@domchristie I've added some assertions to make sure that turbo events are not firing.

Reply:
Nice! An alternative approach, based on https://github.com/hotwired/turbo/blob/bf3c3b0ad56249fa6a3eec3dd59f5d84ec897513/src/tests/helpers/turbo_drive_test_case.ts#L45

```js
  async "test skip link with hash-only path"() {
    const bodyElementId = (await this.body).elementId
    await this.clickSelector('a[href="#main"]')

    await this.nextBeat
    this.assert.equal(await this.pathname, "/src/tests/fixtures/navigation.html")
    this.assert.equal(await this.hash, "#main")
    this.assert.equal((await this.body).elementId, bodyElementId, "does not reload page")
    this.assert.ok(await this.isScrolledToSelector("#main"))

    await this.pressTab()
    const activeElement = await this.remote.getActiveElement()
    const skippedLink = await this.querySelector("#ignored-link")
    const firstLinkWithinMain = await this.querySelector("#main a:first-of-type")
    this.assert.notOk(await activeElement.equals(skippedLink), "skips interactive elements before #main")
    this.assert.ok(await activeElement.equals(firstLinkWithinMain), "skips to first interactive element after #main")
  }
```